### PR TITLE
[SYCLomatic][CMake] Add new match mode "StrictFull" to fix full match not work when '.' follows the matched string

### PR DESCRIPTION
**Pull request by tomflinda**
Signed-off-by: chenwei.sun <chenwei.sun@intel.com>

### CODE REVIEW

**Review by zhiweij1 — APPROVED** [review]

File: clang/lib/DPCT/PatternRewriter.cpp [hunk 1]
```diff
@@ -481,106 +473,53 @@ static std::optional<MatchResult> findFullMatch(const MatchPattern &Pattern,
...
-      }
-
-      std::string ElementContents = Input.substr(Index, Next - Index);
-      if (SrcFileType == SourceFileType::SFT_CMakeScript) {
-        if (Code.Name == "empty" && !ElementContents.empty() &&
-            ElementContents.find_first_not_of(' ') != std::string::npos) {
-          // For reversed variable ${empty}, it should be empty string or string
-          // only including spaces.
+      if (Mode == RuleMatchMode::Full) {
+        if (PatternIndex == 0 && Index - 1 >= 0 &&
```
Comment (AndyCHHuang) [inline clang/lib/DPCT/PatternRewriter.cpp]: the code is similar to full mode can we extract it to a function?

**Review by AndyCHHuang — APPROVED** [review]

File: clang/lib/DPCT/Rules.h [hunk 2]
```diff
@@ -28,7 +28,7 @@ enum RuleKind {
 };
 
 enum RulePriority { Takeover, Default, Fallback };
-enum RuleMatchMode { Partial , Full };
+enum RuleMatchMode { Partial, Full, RawStringMatch };
```
Comment (zhiweij1) [inline clang/lib/DPCT/Rules.h]: How about
```suggestion
enum RuleMatchMode { Partial, Full, StrictFull};
```
?

File: clang/lib/DPCT/PatternRewriter.cpp (hunk 1 above)
Comment (tomflinda) [reply clang/lib/DPCT/PatternRewriter.cpp]: Okay, will update it in the next update.

File: clang/lib/DPCT/Rules.h (hunk 2 above)
Comment (tomflinda) [reply clang/lib/DPCT/Rules.h]: okay

**Review by AndyCHHuang — APPROVED** [review]
LGTM

**Review by zhimingwang36 — APPROVED** [review]

**Pull request merged**